saleor/graphql/checkout/types.py

```diff
@@ -273,6 +277,12 @@ def calculate_available_shipping_methods(data):
                     shipping_method.price = taxed_price.gross
                 else:
                     shipping_method.price = taxed_price.net
+                if not check_shipping_method_for_zip_code(
```

Comment:
IMO, this should be placed in the `applicable_shipping_methods_for_instance`. Then, It will work for order and checkout.  See the PR for excluding products from shipping_method. The case is pretty similar - [shipping/models.py](https://github.com/mirumee/saleor/pull/6506/files#diff-9afc58bf8ebfa6a87d537779f3c333d8310297ad86c01f90adb0c0c9173802c0R143)